l10n_ro_stock_account/models/product_product.py

```diff
@@ -0,0 +1,353 @@
+        currency = company.currency_id
+        # Quantity is negative for out valuation layers.
+        quantity = -1 * quantity
+        vals_list = []
+        vals_tpl = {
+            "product_id": self.id,
+            "value": currency.round(quantity * self.standard_price),
+            "unit_cost": self.standard_price,
+            "quantity": quantity,
+        }
+        if self.cost_method in ("average", "fifo"):
+            fifo_vals_list = self._run_fifo(abs(quantity), company)
+            for fifo_vals in fifo_vals_list:
+                vals = vals_tpl.copy()
+                vals["remaining_qty"] = fifo_vals.get("remaining_qty")
+                # In case of AVCO, fix rounding issue of standard price when needed.
+                if self.cost_method == "average":
+                    rounding_error = currency.round(
+                        self.standard_price * self.quantity_svl - self.value_svl
+                    )
+                    if rounding_error:
+                        # If it is bigger than the (smallest number of the
+                        # currency * quantity) / 2, then it isn't a rounding error
+                        # but a stock valuation error, we shouldn't fix it under the hood ...
+                        if (
+                            abs(rounding_error)
+                            <= (abs(quantity) * currency.rounding) / 2
+                        ):
+                            vals["value"] += rounding_error
+                            vals[
+                                "rounding_adjustment"
+                            ] = "\nRounding Adjustment: %s%s %s" % (
+                                "+" if rounding_error > 0 else "",
+                                float_repr(
+                                    rounding_error,
+                                    precision_digits=currency.decimal_places,
+                                ),
+                                currency.symbol,
+                            )
+                if self.cost_method == "fifo":
+                    vals.update(fifo_vals)
+                vals_list.append(vals)
+        else:
+            vals_list = [vals_tpl]
+        return vals_list
+
+    def _l10n_ro_prepare_domain_fifo(self, domain=None):
+        if not domain:
+            domain = []
+        lot_id = loc_id = None
+        if self._context.get("stock_move_line_id", None):
+            stock_move_line = self._context.get("stock_move_line_id", None)
+
+            if stock_move_line:
+                if isinstance(stock_move_line, int):
+                    stock_move_line = self.env["stock.move.line"].browse(
+                        stock_move_line
+                    )
+                if isinstance(stock_move_line, models.Model):
+                    loc_id = stock_move_line.location_id
+                    lot_id = stock_move_line.lot_id
+        if self.tracking in ["lot", "serial"] and lot_id:
```

Comment:
adaugare optiune in companie pentru descarcarea pe loturi, si daca e pus pe lot doar atunci sa luam in considerare si lotul, default e pe locatie